freeswitch_click2dial/static/src/js/freeswitch_click2dial.js

```diff
@@ -0,0 +1,93 @@
+/*  © 2014-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+    © 2015-2016 Juris Malinens (port to v9)
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).   */
+
+odoo.define('freeswitch_click2dial.click2dial', function (require) {
+"use strict";
+
+var UserMenu = require('web.UserMenu');
+var WebClient = require('web.WebClient');
+var web_client = require('web.web_client');
+var Widget = require('web.Widget');
+var core = require('web.core');
+var _t = core._t;
+
+var click2dial = {};
+
+click2dial.OpenCaller = Widget.extend({
+    template: 'freeswitch_click2dial.OpenCaller',
+
+    start: function () {
+        this.$('#freeswitch-open-caller').on(
+            'click', this.on_open_caller);
+        this._super();
+    },
+
+    on_open_caller: function (event) {
+        event.stopPropagation();
+        var self = this;
+        self.rpc('/freeswitch_click2dial/get_record_from_my_channel', {}).done(function(r) {
+        if (r === false) {
+             self.do_warn(
+                _t('Failure'),
+                _t('Problem in the connection to FreeSWITCH'),
+                false);
+        }
+        else if (typeof r == 'string' && isNaN(r)) {
+             self.do_warn(
+                r,
+                _t('The calling number is not a phone number!'),
+                false);
+        }
+        else if (typeof r == 'string') {
+             var action = {
+                name: _t('Number Not Found'),
+                type: 'ir.actions.act_window',
+                res_model: 'number.not.found',
+                view_mode: 'form',
+                views: [[false, 'form']],
+                target: 'new',
+                context: {'default_calling_number': r},
+             };
+            web_client.action_manager.do_action(action);
+
+            }
+        else if (typeof r == 'object' && r.length == 3) {
+            self.do_notify( // Not working
```

Comment:
On 08/26/2016 01:28 AM, Eric Caudal wrote:
>
> In freeswitch_click2dial/static/src/js/freeswitch_click2dial.js
> <https://github.com/OCA/connector-telephony/pull/101#discussion_r76375235>:
>
> > +        }
> > +        else if (typeof r == 'string') {
> > +             var action = {
> > +                name: _t('Number Not Found'),
> > +                type: 'ir.actions.act_window',
> > +                res_model: 'number.not.found',
> > +                view_mode: 'form',
> > +                views: [[false, 'form']],
> > +                target: 'new',
> > +                context: {'default_calling_number': r},
> > +             };
> > +            web_client.action_manager.do_action(action);
> > +
> > +            }
> > +        else if (typeof r == 'object' && r.length == 3) {
> > +            self.do_notify( // Not working
>
> if not working maybe add a //TODO?
>
This is code that is a direct copy from asterisk_click2dial with just
the renames. I would prefer not to change anything on a line that is
functional as it will make it harder to keep in sync.

Trever